### PR TITLE
goredo: 1.21.0 -> 1.30.0

### PR DESCRIPTION
**Pull request by spacefrogg**
###### Description of changes

[Changelog](http://www.goredo.cypherpunks.ru/News.html#Release-1_005f30_005f0)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/sharness/default.nix [hunk 1]
```diff
@@ -41,11 +41,13 @@ stdenv.mkDerivation rec {
 
   doCheck = true;
 
+  passthru.SHARNESS_TEST_SRCDIR = finalAttrs.finalPackage + "/share/sharness";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sharness/default.nix:44]: Could we not use $out here?

File: pkgs/development/tools/build-managers/goredo/default.nix [hunk 2]
```diff
@@ -50,6 +53,7 @@ buildGoModule rec {
   outputs = [ "out" "info" ];
 
   meta = with lib; {
+    outputsToInstall = [ "out" "info" ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/goredo/default.nix:56]: Do we really need info files all the time?

File: pkgs/development/tools/build-managers/goredo/default.nix [hunk 3]
```diff
@@ -22,12 +22,15 @@ buildGoModule rec {
...
 
-  SHARNESS_TEST_SRCDIR = sharness + "/share/sharness";
+  inherit (sharness) SHARNESS_TEST_SRCDIR;
 
   vendorSha256 = null;
   subPackages = [ "." ];
 
-  preBuild = "cd src";
+  preBuild = ''
+    export GOPATH=$TMPDIR/go/bin
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/goredo/default.nix:31]: That should already exist. What problem did you try to solve by setting GOPATH?

File: pkgs/development/tools/build-managers/goredo/default.nix (hunk 2 above)
Comment (spacefrogg) [reply pkgs/development/tools/build-managers/goredo/default.nix:56]: The info file is the only form of documentation. Should I leave it out?

File: pkgs/development/libraries/sharness/default.nix (hunk 1 above)
Comment (spacefrogg) [reply pkgs/development/libraries/sharness/default.nix:44]: First, being a passthru and not being a shell variable, I would suspect that $out is just wrong here. Second, this variable must point to the resulting store path after all overrides have been applied. Overriding sharness with augmented test functions belongs to the package's API.

File: pkgs/development/tools/build-managers/goredo/default.nix (hunk 3 above)
Comment (spacefrogg) [reply pkgs/development/tools/build-managers/goredo/default.nix:31]: It does not exist. It used to but is not set anymore. Without setting GOPATH, the buildPhase just does nothing.

I am not a go packaging expert, but I suspect that the packages does not adhere to the packaging standards. Setting GOPATH again (like it used to) solves the issue.

File: pkgs/development/libraries/sharness/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/sharness/default.nix:44]: Right, I think we also can't use placeholder 🤔

File: pkgs/development/tools/build-managers/goredo/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/build-managers/goredo/default.nix:56]: outputsToInstall should only install the info output if the program is not working without it. If users opt out of installing info files then they should not be enforced here

File: pkgs/development/tools/build-managers/goredo/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/build-managers/goredo/default.nix:31]: It is set here https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/go/module.nix#L107

File: pkgs/development/tools/build-managers/goredo/default.nix (hunk 2 above)
Comment (spacefrogg) [reply pkgs/development/tools/build-managers/goredo/default.nix]: Right, I removed it.

File: pkgs/development/tools/build-managers/goredo/default.nix (hunk 3 above)
Comment (spacefrogg) [reply pkgs/development/tools/build-managers/goredo/default.nix]: Thanks! I don't know how I could miss that... Now, that I'm using it right, it's even simpler than before.

**Comment by spacefrogg** [comment]
I just snuck in a sharness update, because the version in the package is no longer maintained.

File: pkgs/development/libraries/sharness/default.nix [hunk 4]
```diff
@@ -6,15 +6,15 @@
...
+  version = "1.2.0";
 
   src = fetchFromGitHub {
-    owner = "chriscool";
-    repo = pname;
-    rev = "3f238a740156dd2082f4bd60ced205e05894d367"; # 2020-12-09
-    sha256 = "FCYskpIqkrpNaWCi2LkhEkiow4/rXLe+lfEWNUthLUg=";
+    owner = "felipec";
+    repo = "sharness";
+    rev = "v1.2.0";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sharness/default.nix:16]:
```suggestion
    rev = "v${finalAttrs.version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**